app/controlplane/integrations/dependencytrack/cyclonedx/v1/api.proto

```diff
@@ -22,28 +22,19 @@ import "validate/validate.proto";
 
 // RegistrationRequest is what the integration expect as input of the pre-registration step
```

Comment:
Do we follow any naming convention here dependencytrack/cyclonedx/v1? Is "dependencytrack" the integration name and "cyclonedx" a feature available in the extension? I know this will change but wondering what is the current approach that we follow here.

Reply:
Good question.

I actually followed a similar approach [than the one at Kubeapps](https://github.com/vmware-tanzu/kubeapps/tree/main/cmd/kubeapps-apis/plugins) which seems to be `{solution}/{purpose}/{version}`. I replaced `purpose` for `input type`.

That said, I do not think this pattern works anymore since an extension might be subscribed to more than one `material type` or to both materials and attestations

In fact, after our discussion offline this morning, where we laid of some potential extension `roles`, it might make sense to in fact do something like

`{role}/{provider}/{version}` for example

- `fanout/dependencytrack/v1`
- `fanout/oci/v1`

What do you think? @danlishka ?

Reply:
I would make it flat for now, and I think it makes more sense to rename integrations to extensions.